kratos/sources/model_part_io.cpp

```diff
@@ -1775,6 +1776,67 @@ namespace Kratos
+                if(is_included == variables.end()){
+                    variables.insert(var.first->Name());
+                    // determine variable type
+                    if(KratosComponents<Variable<bool>>::Has(var.first->Name())){
+                        Variable<bool> variable = KratosComponents<Variable<bool>>::Get(var.first->Name());
+                        WriteElementalDataBlock<Variable<bool>>(rThisElements, variable);
+                    }
+                    else if(KratosComponents<Variable<int>>::Has(var.first->Name())){
+                        Variable<int> variable = KratosComponents<Variable<int>>::Get(var.first->Name());
+                        WriteElementalDataBlock<Variable<int>>(rThisElements, variable);
+                    }
+                    else if(KratosComponents<Variable<double>>::Has(var.first->Name())){
+                        Variable<double> variable = KratosComponents<Variable<double>>::Get(var.first->Name());
+                        WriteElementalDataBlock<Variable<double>>(rThisElements, variable);
+                    }
+                    else if(KratosComponents<array_1d_component_type>::Has(var.first->Name())){
+                        array_1d_component_type variable = KratosComponents<array_1d_component_type>::Get(var.first->Name());
+                        WriteElementalDataBlock<array_1d_component_type>(rThisElements, variable);
+                    }
+                    else if(KratosComponents<Variable<array_1d<double,3>>>::Has(var.first->Name())){
+                        Variable<array_1d<double,3>> variable = KratosComponents<Variable<array_1d<double,3>>>::Get(var.first->Name());
+                        WriteElementalDataBlock<Variable<array_1d<double,3>>>(rThisElements, variable);
+                    }
+                    else if(KratosComponents<Variable<Quaternion<double>>>::Has(var.first->Name())){
+                        Variable<Quaternion<double>> variable = KratosComponents<Variable<Quaternion<double>>>::Get(var.first->Name());
+                        WriteElementalDataBlock<Variable<Quaternion<double>>>(rThisElements, variable);
+                    }
+                    else if(KratosComponents<Variable<Vector>>::Has(var.first->Name())){
+                        Variable<Vector> variable = KratosComponents<Variable<Vector>>::Get(var.first->Name());
+                        WriteElementalDataBlock<Variable<Vector>>(rThisElements, variable);
+                    }
+                    else if(KratosComponents<Variable<Matrix>>::Has(var.first->Name())){
+                        Variable<Matrix> variable = KratosComponents<Variable<Matrix>>::Get(var.first->Name());
+                        WriteElementalDataBlock<Variable<Matrix>>(rThisElements, variable);
+                    }
+                    else
+                        std::cout << var.first->Name() << " is not a valid variable for output!!!" << std::endl;
```

Comment:
Can we replace this for the new logger?

Reply:
In fact we can do it for the whole file...

Reply:
Can you do that? I don't know about the new logger, just copied the code from the nodal data. Or you can explain it to me when we see each other again.

Reply:
It is a new thing in fact, @pooyan-dadvand do you thing that is a good idea to do in this same PR or to do a new PR in purpose for the logger?

Besides:
https://github.com/KratosMultiphysics/Kratos/wiki/How-to-use-Logger

Reply:
I would go for a separate PR.